nixos/modules/services/network-filesystems/nfsd.nix

```diff
@@ -1,175 +1,166 @@
+        Default is all.  See the <option>-H</option> option in
+        <citerefentry><refentrytitle>nfsd</refentrytitle>
+        <manvolnum>8</manvolnum></citerefentry>.
+      '';
+    };
 
-        lockdPort = mkOption {
-          type = types.nullOr types.int;
-          default = null;
-          example = 4001;
-          description = ''
-            Use a fixed port for the NFS lock manager kernel module
-            (<literal>lockd/nlockmgr</literal>).  This is useful if the
-            NFS server is behind a firewall.
-          '';
-        };
+    nproc = mkOption {
+      type = types.int;
+      default = 8;
+      description = ''
+        Number of NFS server threads.  Defaults to the recommended value of 8.
+      '';
+    };
 
-        statdPort = mkOption {
-          type = types.nullOr types.int;
-          default = null;
-          example = 4000;
-          description = ''
-            Use a fixed port for <command>rpc.statd</command>. This is
-            useful if the NFS server is behind a firewall.
-          '';
-        };
+    createMountPoints = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to create the mount points in the exports file at startup time.";
+    };
 
-      };
+    mountdPort = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      example = 4002;
+      description = ''
+        Use fixed port for rpc.mountd, useful if server is behind firewall.
+      '';
+    };
 
+    lockdPort = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      example = 4001;
+      description = ''
+        Use a fixed port for the NFS lock manager kernel module
+        (<literal>lockd/nlockmgr</literal>).  This is useful if the
+        NFS server is behind a firewall.
+      '';
     };
 
+    statdPort = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      example = 4000;
+      description = ''
+        Use a fixed port for <command>rpc.statd</command>. This is
+        useful if the NFS server is behind a firewall.
+      '';
+    };
   };
 
 
   ###### implementation
 
   config = mkIf cfg.enable {
 
-    services.nfs.extraConfig = ''
-      [nfsd]
-      threads=${toString cfg.nproc}
-      ${optionalString (cfg.hostName != null) "host=${cfg.hostName}"}
-      ${cfg.extraNfsdConfig}
-
-      [mountd]
-      ${optionalString (cfg.mountdPort != null) "port=${toString cfg.mountdPort}"}
-
-      [statd]
-      ${optionalString (cfg.statdPort != null) "port=${toString cfg.statdPort}"}
-
-      [lockd]
-      ${optionalString (cfg.lockdPort != null) ''
-        port=${toString cfg.lockdPort}
-        udp-port=${toString cfg.lockdPort}
-      ''}
-    '';
-
-    services.rpcbind.enable = true;
+    services = {
+      nfs.settings = recursiveUpdate
+        {
+          nfsd = { threads = cfg.nproc; }
+            // optionalAttrs (cfg.hostName != null) { host = cfg.hostName; };
```

Comment:
Breaking backwards compatibility isn't uncommon, but I'm not sure what the exact procedure is.

Reply:
The main reason is really that sometimes options from an RFC42 modules are still promoted to the top level in order to ensure it can be easily referenced from elsewhere, for improved type checking and to easily apply a function to its contents or just any kind of special handling. 

We can always deprecate these if it makes sense - I just haven't really looked at if they should go from here.

Reply:
How would we declare the options under something declared as `(pkgs.formats.ini).type`?

Reply:
Boy, do I feel dumb now..... Thanks man, will fix.